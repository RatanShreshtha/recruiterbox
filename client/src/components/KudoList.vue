<template>
  <b-row cols="3">
    <b-col></b-col>
    <b-col>
      <h1 class="text-center" v-if="kudos.length > 0">{{ heading }}</h1>
      <h1 class="text-center" v-else>{{ placeholder }}</h1>
    </b-col>
    <b-col></b-col>
    <b-col
      v-for="kudo in kudos"
      :key="kudo.id"
      class="my-2"
      style="display: flex"
    >
      <b-card
        :title="kudo.title"
        header-tag="header"
        footer-tag="footer"
        class="h-100 w-100"
      >
        <b-card-text>
          {{ kudo.description | truncate(100) }}
          <router-link :to="{ name: 'KudoDetail', params: { kudoId: kudo.id } }"
            >See More</router-link
          >
        </b-card-text>
        <template v-slot:footer
          >Awarded: {{ kudo.awarded | moment("calendar") }}</template
        >
      </b-card>
    </b-col>
    <b-col></b-col>
    <b-col>
      <b-pagination-nav
        :link-gen="linkGen"
        :number-of-pages="pages"
        align="center"
        class="mt-2"
        use-router
        v-if="kudos.length > 0"
      ></b-pagination-nav>
    </b-col>
    <b-col></b-col>
  </b-row>
</template>

<script>
import { mapState } from "vuex";

import { apiService } from "@/service";
export default {
  name: "KudoList",
  props: {
    listingType: {
      type: String,
      required: true
    },
    heading: {
      type: String
    },
    placeholder: {
      type: String
    }
  },
  computed: {
    ...mapState(["id", "token"]),
    page() {
      return this.$route.query.page || 1;
    }
  },
  watch: {
    page: {
      immediate: true,
      async handler() {
        this.getKudos();
      }
    }
  },
  data() {
    return {
      kudos: [],
      pages: 1,
      perPage: 9
    };
  },
  async mounted() {
    this.getKudos();
  },
  methods: {
    linkGen(pageNum) {
      return pageNum === 1 ? "?" : `?page=${pageNum}`;
    },
    async getKudos() {
      try {
        const resp = await apiService.get(
          `/api/kudos/?${this.listingType}=${this.id}&page=${this.page}`,
          {
            headers: {
              Authorization: `Token ${this.token}`
            }
          }
        );
        this.kudos = resp.data.results;
        this.pages = resp.data.count / this.perPage;
      } catch (error) {
        console.log("error", error);
      }
    }
  }
};
</script>

<style scoped></style>
