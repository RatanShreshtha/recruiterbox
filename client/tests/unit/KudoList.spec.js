import { shallowMount, createLocalVue } from "@vue/test-utils";

import { createRenderer } from "vue-server-renderer";

import Vuex from "vuex";
import VueRouter from "vue-router";
import { BootstrapVue, IconsPlugin } from "bootstrap-vue";
import createPersistedState from "vuex-persistedstate";

import KudoList from "@/components/KudoList.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

localVue.use(BootstrapVue);
localVue.use(IconsPlugin);

describe("KudoList.vue", () => {
  let propsData;
  let state;
  let store;
  let routes;
  let router;

  beforeEach(() => {
    propsData = {
      listingType: "by_user"
    };
    state = {
      id: null,
      kudos: 0,
      username: null,
      first_name: null,
      last_name: null,
      email: null,
      token: null
    };

    store = new Vuex.Store({
      plugins: [createPersistedState()],
      state
    });

    routes = [
      {
        path: "/",
        name: "Home"
      }
    ];

    router = new VueRouter({
      routes
    });
  });
  it("matches snapshot", () => {
    const renderer = createRenderer();

    const wrapper = shallowMount(KudoList, {
      localVue,
      store,
      propsData,
      router
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
