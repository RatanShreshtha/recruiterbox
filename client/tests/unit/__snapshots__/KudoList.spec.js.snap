// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KudoList.vue matches snapshot 1`] = `
<b-row-stub tag="div" cols="3">
  <b-col-stub tag="div"></b-col-stub>
  <b-col-stub tag="div">
    <h1 class="text-center"></h1>
  </b-col-stub>
  <b-col-stub tag="div"></b-col-stub>
  <b-col-stub tag="div"></b-col-stub>
  <b-col-stub tag="div">
    <!---->
  </b-col-stub>
  <b-col-stub tag="div"></b-col-stub>
</b-row-stub>
`;
